diff --git a/quantum/process_keycode/process_combo.c b/quantum/process_keycode/process_combo.c
--- a/quantum/process_keycode/process_combo.c
+++ b/quantum/process_keycode/process_combo.c
@@ -330,6 +330,9 @@ void drop_combo_from_buffer(uint16_t combo_index) {
     }
 }
 
+__attribute__((weak)) void combo_triggered_user(uint16_t combo_index, uint16_t keycode) {
+}
+
 void apply_combo(uint16_t combo_index, combo_t *combo) {
     /* Apply combo's result keycode to the last chord key of the combo and
      * disable the other keys. */
@@ -371,6 +374,7 @@ void apply_combo(uint16_t combo_index, combo_t *combo) {
 
             qrecord->combo_index = combo_index;
             ACTIVATE_COMBO(combo);
+            combo_triggered_user(combo_index, combo->keycode);
 
             break;
         } else {
diff --git a/quantum/process_keycode/process_combo.h b/quantum/process_keycode/process_combo.h
--- a/quantum/process_keycode/process_combo.h
+++ b/quantum/process_keycode/process_combo.h
@@ -66,6 +66,7 @@ bool process_combo(uint16_t keycode, keyrecord_t *record);
 void combo_task(void);
 void process_combo_event(uint16_t combo_index, bool pressed);
+void combo_triggered_user(uint16_t combo_index, uint16_t keycode);
 
 void combo_enable(void);
 void combo_disable(void);
